src/main/java/core/basesyntax/NumberSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class NumberSupplier {
```

Comment:
I'd say that the `Random` class is a number supplier by itself, so we can just use it straight without this class.
Plus this was not part of the task so let's refactor your solution just a bit, although I like your effort to do a clear separation of responsibility :+1: